File: scripts/run-migration.mjs
import { spawn } from 'child_process';
import { fileURLToPath } from 'url';
import { dirname } from 'path';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

console.log('[Migration] Starting Prisma migration...');

const prisma = spawn('npx', ['prisma', 'migrate', 'deploy'], {
  cwd: dirname(__dirname),
  stdio: 'inherit'
});

prisma.on('close', (code) => {
  if (code === 0) {
    console.log('[Migration] Migration completed successfully!');
    process.exit(0);
  } else {
    console.error(`[Migration] Migration failed with code ${code}`);
    process.exit(1);
  }
});

prisma.on('error', (err) => {
  console.error('[Migration] Error running migration:', err);
  process.exit(1);
});

File: scripts/run_migration.py
import subprocess
import sys
import os

os.chdir('/vercel/share/v0-project')

# Run Prisma migration
print("[v0] Running Prisma database migration...")
result = subprocess.run([sys.executable, '-m', 'pip', 'install', 'prisma'], capture_output=True)
print("[v0] Prisma install result:", result.returncode)

# Run the actual migration using npx
result = subprocess.run(['npx', 'prisma', 'migrate', 'deploy'], capture_output=True, text=True)
print("[v0] Migration stdout:", result.stdout)
print("[v0] Migration stderr:", result.stderr)
print("[v0] Migration result code:", result.returncode)

if result.returncode == 0:
    print("[v0] Migration completed successfully!")
else:
    print("[v0] Migration failed with return code:", result.returncode)
    sys.exit(1)
